wkcuber/wkcuber/export_wkw_as_nifti.py

```diff
@@ -0,0 +1,201 @@
+        "x,y,z is the offset of the wkw layer into the final bounding box; "
+        "width,height,depth corresponds to final dimensions",
+        default=None,
+        type=parse_bounding_box,
+    )
+
+    parser.add_argument(
+        "--cropping_bbox",
+        help="After reading (and potentially padding) the input data, the output is cropped according to --cropping_bbox. --cropping_bbox must be contained in --padded_bbox"
+        "Format is: x,y,z,width,height,depth",
+        default=None,
+        type=parse_bounding_box,
+    )
+
+    add_verbose_flag(parser)
+    add_distribution_flags(parser)
+
+    return parser
+
+
+def export_layer_to_nifti(
+    wkw_file_path: Path,
+    source_bbox: BoundingBox,
+    mag: Mag,
+    layer_name: str,
+    destination_path: Path,
+    name: str,
+    padded_bbox: Optional[BoundingBox] = None,
+    cropping_bbox: Optional[BoundingBox] = None,
+) -> None:
+    dataset = Dataset(wkw_file_path)
+    layer = dataset.get_layer(layer_name)
+    mag_layer = layer.get_mag(mag)
+
+    is_segmentation_layer = layer.category == LayerCategories.SEGMENTATION_TYPE
+
+    data = mag_layer.read(source_bbox.topleft, source_bbox.size)
+    data = data.transpose(1, 2, 3, 0)
+    logging.info(f"Shape with layer {data.shape}")
+
+    data = np.array(data)
+    if is_segmentation_layer and data.max() > 0:
+        factor = np.iinfo("uint8").max / data.max()
+        data = data * factor
+        data = data.astype(np.dtype("uint8"))
+
+    if padded_bbox:
+        data_bbox_relative_to_pad_bbox = BoundingBox(
+            padded_bbox.topleft, data.shape[:3]
+        )
+        pad_bbox_as_origin = BoundingBox((0, 0, 0), padded_bbox.size)
+
+        assert pad_bbox_as_origin.contains_bbox(
+            data_bbox_relative_to_pad_bbox
+        ), "padded_bbox should contain source_bbox"
+
+        padding_per_axis = []
+
+        for i in range(3):
+            if padded_bbox.size[i] == data.shape[i]:
+                padding_per_axis.append((0, 0))
+            else:
+                left_pad = data_bbox_relative_to_pad_bbox.topleft[i]
+                right_pad = (
+                    pad_bbox_as_origin.size[i]
+                    - data_bbox_relative_to_pad_bbox.bottomright[i]
+                )
+                padding_per_axis.append((left_pad, right_pad))
+
+        padding_per_axis.append((0, 0))
+        data = np.pad(data, padding_per_axis, mode="constant", constant_values=0)
+
+        if cropping_bbox is not None:
+            assert pad_bbox_as_origin.contains_bbox(
+                cropping_bbox
+            ), "padded_bbox should contain cropping_bbox"
+
+            logging.info(f"Using Bounding Box {cropping_bbox}")
+
+            data = data[
+                cropping_bbox.topleft[0] : cropping_bbox.bottomright[0],
+                cropping_bbox.topleft[1] : cropping_bbox.bottomright[1],
+                cropping_bbox.topleft[2] : cropping_bbox.bottomright[2],
+            ]
```

Comment:
```suggestion
            data = cropping_bbox.slice_array(data)
```

Reply:
This means that the cropping bbox is relative to the padded data. I would have thought it would be easier to use if all three bounding boxes are specified in the same global coordinate system by the user (see my mental model picture in the other comment). But maybe I'm wrong and it's easier to use the way you designed it. If you stick to that semantics, please update the argparse help to reflect the relativity. Also, the comments about the containment-relationships are then incorrect (and not useful) probably.